common/src/main/java/org/opensearch/ml/common/connector/functions/preprocess/ConnectorPreProcessFunction.java

```diff
@@ -46,6 +47,10 @@ public void validateTextDocsInput(MLInput mlInput) {
         if (!(mlInput.getInputDataset() instanceof TextDocsInputDataSet)) {
             throw new IllegalArgumentException("This pre_process_function can only support TextDocsInputDataSet");
```

Comment:
can you make error message more meaningful to the user, if possible please add details of how/what they need to change to fix it.
in addition to that we can log more details at `warn` or `info` level, like what's the actual class of input dataset object

Reply:
It make sense we log details of input dataset type, but for changing the error message giving an TextDocsInputDataSet example doesn't seem make sense because the format of the error message will be odd. I prefer we give a link to the source code of TextDocsInputDataSet and user can learn the data structure in the code. WDYT?

Reply:
Is there any tutorial / website link we can refer to instead of source code link?

Reply:
We don't have a good document describing this, I think it's fine to add the key data structure of this request to the error message. Will make the change soon.